### DISCUSSION TITLE
Checkbox focus adorner not visible in dark theme variant

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#16982] [Q&A] [answered]
With a light theme variant selected, a  checkbox control is drawn with a dashed line around it when the focus is moved to it using the tab-key.
The dashed line is not visible when using a dark theme variant, however.

How can I make it show?
Preferably  in a whole-sale way for other controls as well.
I am using the Simple theme.

**Comment by stevemonaco** [comment, marked as answer]
You need to replace / customize the `DefaultFocusAdorner`. It's hardcoded to Black, so you'll need to use a theme-friendly resource. The focus adorner is present in Dark Theme, but obviously very hard to see. Might be worth a PR or issue to fix.

https://github.com/AvaloniaUI/Avalonia/blob/dbbb6d1819222eeeefbb746e695709c288954529/src/Avalonia.Themes.Simple/Controls/AdornerLayer.xaml#L4-L13

eg. Via selector, but could replace the `ControlTheme` instead:
![image](https://github.com/user-attachments/assets/defd50b9-d196-42e5-b8a8-a96323db5982)

**Comment by wgraaf** [comment]
Ended up with this in app.xaml <Application>:

```
  <Application.Styles>
...
    <Style Selector=":is(Control)">
      <Setter Property="FocusAdorner">
        <FocusAdornerTemplate>
          <Rectangle StrokeDashArray="1, 2" Stroke="{DynamicResource FocusAdornerStroke}" StrokeThickness="2" Margin="1"/>
        </FocusAdornerTemplate>
      </Setter>
    </Style>
...
  </Application.Styles>

```
And

```
  <Application.Resources>
    <ResourceDictionary>
      <ResourceDictionary.ThemeDictionaries>
        <ResourceDictionary x:Key="Light">
...
          <SolidColorBrush x:Key='FocusAdornerStroke'>RGB(180,180,180)</SolidColorBrush>
...
        </ResourceDictionary>
        <ResourceDictionary x:Key="Dark">
...
          <SolidColorBrush x:Key='FocusAdornerStroke'>RGB(160,160,160)</SolidColorBrush>
...
        </ResourceDictionary>
      </ResourceDictionary.ThemeDictionaries>
    </ResourceDictionary>
  </Application.Resources>

```
